src/components/AppShell.jsx

```diff
@@ -0,0 +1,129 @@
+import {
+  AppShell,
+  Burger,
+  Footer,
+  Header,
+  MediaQuery,
+  Menu,
+  Navbar,
+  Stack,
+  Text,
+  UnstyledButton,
+  useMantineTheme,
+} from "@mantine/core";
+import {
+  IconLogout,
+  IconMoon,
+  IconSettings,
+  IconSunFilled,
+  IconSwitchHorizontal,
+  IconUserCircle,
+} from "@tabler/icons-react";
+import { useState } from "react";
+import { DemoComponent } from "./DemoComponent/DemoComponent";
+
+export default function AppShellDemo() {
+  const theme = useMantineTheme();
+  const [opened, setOpened] = useState(false);
+  const [userMenuOpened, setUserMenuOpened] = useState(false);
+
+  return (
+    <AppShell
+      styles={{
+        main: {
+          background:
+            theme.colorScheme === "dark"
+              ? theme.colors.dark[8]
+              : theme.colors.gray[0],
+        },
+      }}
+      header={
+        <Header height={{ base: 50, md: 70 }} p="md">
+          <div
+            style={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "space-between",
+              height: "100%",
+            }}
+          >
+            <MediaQuery largerThan="sm" styles={{ display: "none" }}>
+              <Burger
+                opened={opened}
+                onClick={() => setOpened((o) => !o)}
```

Comment:
Move all logic above: create a function(-s) for opening and closing the`Burger`

Reply:
Also, that's the only place to use this function, do we really need to create function for that?

Reply:
Done.